rafthttp/http.go

```diff
@@ -117,11 +130,26 @@ func (h *streamHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	fromStr := strings.TrimPrefix(r.URL.Path, RaftStreamPrefix+"/")
+	var t streamType
+	switch path.Dir(r.URL.Path) {
+	case path.Join(RaftStreamPrefix, string(streamTypeMsgApp)):
+		t = streamTypeMsgApp
+	case path.Join(RaftStreamPrefix, string(streamTypeMessage)):
+		t = streamTypeMessage
+	// backward compatibility
+	case RaftStreamPrefix:
+		t = streamTypeMsgApp
+	default:
+		log.Printf("rafthttp: path %s cannot be parsed", r.URL.Path)
+		http.Error(w, "invalid path", http.StatusNotFound)
+		return
+	}
+
+	fromStr := path.Base(r.URL.Path)
 	from, err := types.IDFromString(fromStr)
 	if err != nil {
-		log.Printf("rafthttp: path %s cannot be parsed", fromStr)
-		http.Error(w, "invalid path", http.StatusNotFound)
+		log.Printf("rafthttp: from %s cannot be parsed", fromStr)
```

Comment:
this kind of logging message is not very helpful. we, at lease, should explicitly tell when it cannot be parsed and why